libsolidity/interface/CompilerStack.cpp

```diff
@@ -508,23 +508,35 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 		return std::equal(_a.begin(), _a.end(), _b.begin());
 	};
 
+	using filepath = boost::filesystem::path;
 	size_t longestPrefix = 0;
+	size_t longestContext = 0;
 	string longestPrefixTarget;
+
 	for (auto const& redir: m_remappings)
 	{
-		// Skip if we already have a closer match.
-		if (longestPrefix > 0 && redir.prefix.length() <= longestPrefix)
+		filepath redirContext(redir.context);
+		filepath redirPrefix(redir.prefix);
+		string contextFileString = redirContext.generic_string();
```

Comment:
Arguably, `myDir` could also perform a remapping on `myDirectory`, but I guess it is better to restrict both the prefix and the context to directories only. Could you write a helper function to do that?